langworld_db_data/removers/feature_remover.py

```diff
@@ -0,0 +1,311 @@
+
+        category_id_where_feature_is_removed = extract_category_id(feature_id)
+
+        rows_with_removed_rows_and_updated_value_indices = (
+            self._update_indices_after_given_line_number_if_necessary(
+                match_column_name="id",
+                match_content=category_id_where_feature_is_removed,
+                id_type_that_must_be_updated="value",
+                index_type_that_must_be_updated="feature",
+                line_number_after_which_rows_must_be_updated=line_number_of_first_removed_value,
+                rows=rows_with_removed_rows,
+            )
+        )
+
+        rows_with_removed_rows_and_updated_feature_and_value_indices = (
+            self._update_indices_after_given_line_number_if_necessary(
+                match_column_name="feature_id",
+                match_content=category_id_where_feature_is_removed,
+                id_type_that_must_be_updated="feature",
+                index_type_that_must_be_updated="feature",
+                line_number_after_which_rows_must_be_updated=line_number_of_first_removed_value,
+                rows=rows_with_removed_rows_and_updated_value_indices,
+            )
+        )
+
+        write_csv(
+            rows=rows_with_removed_rows_and_updated_feature_and_value_indices,
+            path_to_file=self.output_file_with_listed_values,
+            overwrite=True,
+            delimiter=",",
+        )
+
+    def _remove_from_feature_profiles(
+        self,
+        feature_id: str,
+    ) -> None:
+
+        feature_profiles = self.input_dir_with_feature_profiles.glob("*.csv")
+
+        for feature_profile in feature_profiles:
+
+            rows = read_dicts_from_csv(feature_profile)
+
+            rows_with_removed_row, line_number_of_removed_row = (
+                self._remove_one_matching_row_and_return_its_line_number(
+                    match_column_name="feature_id", match_content=feature_id, rows=rows
+                )
+            )
+
+            rows_with_removed_row_and_updated_indices = (
+                self._update_indices_after_given_line_number_if_necessary(
+                    match_column_name="feature_id",
+                    match_content=extract_category_id(feature_id),
+                    id_type_that_must_be_updated="feature",
+                    index_type_that_must_be_updated="feature",
+                    line_number_after_which_rows_must_be_updated=line_number_of_removed_row,
+                    rows=rows_with_removed_row,
+                    rows_are_a_feature_profile=True,
+                )
+            )
+
+            write_csv(
+                rows=rows_with_removed_row_and_updated_indices,
+                path_to_file=self.output_dir_with_feature_profiles / feature_profile.name,
+                overwrite=True,
+                delimiter=",",
+            )
+
+    @staticmethod
+    def _remove_one_matching_row_and_return_its_line_number(
```

Comment:
I think this method can be made a bit more universal and put into `tools.files.csv_xls`
It will definitely come in handy.
Maybe this method would work at a higher level, getting path to CSV file and amending the file by removing a row.